### PR TITLE
Quicken DirectStoreMuxer.idle

### PR DESCRIPTION
**Pull request by ianull**
Observed a tons of DirectStoreMuxer.idle arrived while adding participants consecutively ends up
client apis `store` is blocked for a big variant time due to limited threads on the service end and
much contention on the lock `storeMutex` which also blocks other critical paths i.e. creating / evicting stores.

Work with #5928 to achieve much better write performance. (1.5s to 25 ms per participant (persistent)).

### CODE REVIEW

File: java/arcs/core/storage/DirectStoreMuxer.kt [hunk 1]
```diff
@@ -75,12 +75,12 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
...
-        stores.values.map {
-            withContext(coroutineContext) {
-                launch { it.store.idle() }
-            }
-        }.joinAll()
-    }
+        stores.values.toList()
+    }.map {
+        withContext(coroutineContext) {
+            launch { it.store.idle() }
```
Comment (jblebrun) [inline java/arcs/core/storage/DirectStoreMuxer.kt:81]: Is there any benefit to launching these in parallel if the goal is just to wait for all to complete?

Comment (ianull) [reply java/arcs/core/storage/DirectStoreMuxer.kt:81]: It depends on the overhead of `store.idle()`.
If the overhead of each `store.idle()` is bigger than an `launch` overhead, the launch here can help more or less.
If not, the launch here would hurt instead.

Just take a look at the implementation of `store.idle()`, it does nothing at this moment...
So we can remove the multiple launches to get some time back.
Nice find!

Comment (ianull) [reply java/arcs/core/storage/DirectStoreMuxer.kt:81]: btw, the PR improves writes for all "persistent" handles (I knew we changed to memdb for pexxx) a lot from 1+ second to around 30 ms (100 participants).

Comment (ianull) [reply java/arcs/core/storage/DirectStoreMuxer.kt]: Removed launches and added comments.

File: java/arcs/core/storage/DirectStoreMuxer.kt [hunk 2]
```diff
@@ -75,11 +71,14 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
...
-            }
-        }.joinAll()
+        stores.values.toList()
+    }.map {
+        /**
+         * If the overhead/wall-time of [DirectStore.idle] is longer than an
+         * [CoroutineScope.launch] i.e. more than 5ms debounce time, launching
+         * [DirectStore.idle]s in parallel can further help performance,
+         */
+        it.store.idle()
```
Comment (cromwellian) [inline java/arcs/core/storage/DirectStoreMuxer.kt:81]: Doesn't this need a launch {} to make it parallel? Or is this comment a TODO?

Comment (ianull) [reply java/arcs/core/storage/DirectStoreMuxer.kt:81]: Currently `store.idle` does nothing so `launch` here become negative performance impact.
So I put the comment there.

**Pull request merged**